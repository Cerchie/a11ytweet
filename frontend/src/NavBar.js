import * as React from 'react'
import { useContext } from 'react'
import { NavLink, Link } from 'react-router-dom'
import UserContext from './UserContext'
// TODO-- ADD LOGIC SO PROFILE ONLY APPEARS WHEN LOGGED IN
function NavBar({ logout }) {
    if (useContext(UserContext) !== undefined) {
        const { currentUser } = useContext(UserContext)

        console.debug('Navigation', 'currentUser=', currentUser)

        function loggedOutNav() {
            return (
                <div>
                    Welcome to A11y!
                    <div>
                        <Link to="/login">Login</Link>
                    </div>
                    <div>
                        <Link to="/signup">Signup</Link>
                    </div>
                    <div>
                        <Link to="/">Home</Link>
                    </div>
                    <div>
                        <Link to="/links">a11y repos</Link>
                    </div>
                </div>
            )
        }

        function loggedInNav() {
            return (
                <>
                    <Link exact to="/" className="navbar-brand">
                        Welcome {currentUser.username}!
                    </Link>
                    <nav className="ml-auto" navbar>
                        <p>Hi {currentUser.username}!</p>
                        <div>
                            <Link to="/links">a11y repos</Link>
                        </div>
                        <div>
                            <Link to="/profile">Edit Profile</Link>
                        </div>
                        <div>
                            <Link to="/links">Your Links</Link>
                        </div>
                        <div>
                            <Link className="nav-link" to="/" onClick={logout}>
                                Log out{' '}
                                {currentUser.first_name || currentUser.username}
                            </Link>
                        </div>
                    </nav>
                </>
            )
        }

        return (
            <nav className="Navigation navbar navbar-expand-md">
                <Link className="navbar-brand" to="/">
                    A11yTweet
                </Link>
                {currentUser ? loggedInNav() : loggedOutNav()}
            </nav>
        )
    }
    return <div> No navbar. </div>
}

export default NavBar

//code adapted from my Springboard projecthttps://github.com/Cerchie/react-jobly/blob/main/frontend/src/NavBar.js
