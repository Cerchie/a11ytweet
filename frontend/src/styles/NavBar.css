li {
    margin: 1rem;
}

h1 {
    margin: 1rem;
}

ul {
    list-style-type: none;
}

:focus {
    outline: 3px solid blue;
    padding: 0.25rem;
}
