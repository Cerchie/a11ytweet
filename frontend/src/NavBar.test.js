import { render, screen } from '@testing-library/react'
import React from 'react'
import NavBar from './NavBar'
import { BrowserRouter } from 'react-router-dom'

test('renders learn react link', () => {
    render(
        <BrowserRouter>
            <NavBar />
        </BrowserRouter>
    )
})
