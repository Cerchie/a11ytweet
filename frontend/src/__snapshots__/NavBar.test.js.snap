// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      A11yTweet
    </a>
    <a
      class="navbar-brand"
      href="/"
    >
      Welcome testuser!
    </a>
    <nav
      class="ml-auto"
    >
      <p>
        Hi testuser!
      </p>
      <div>
        <a
          href="/links"
        >
          a11y repos
        </a>
      </div>
      <div>
        <a
          href="/profile"
        >
          Edit Profile
        </a>
      </div>
      <div>
        <a
          href="/links"
        >
          Your Links
        </a>
      </div>
      <div>
        <a
          class="nav-link"
          href="/"
        >
          Log out testuser
        </a>
      </div>
    </nav>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      A11yTweet
    </a>
    <div>
      Welcome to A11y!
      <div>
        <a
          href="/login"
        >
          Login
        </a>
      </div>
      <div>
        <a
          href="/signup"
        >
          Signup
        </a>
      </div>
      <div>
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div>
        <a
          href="/links"
        >
          a11y repos
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
